src/shared/components/axisTicks/TimeTickInput.tsx

```diff
@@ -0,0 +1,236 @@
+// Libraries
+import React, {
+  ChangeEvent,
+  createRef,
+  CSSProperties,
+  FC,
+  RefObject,
+  useState,
+} from 'react'
+
+// Utils
+import {useOneWayState} from 'src/shared/utils/useOneWayState'
+import {convertUserInputValueToNumOrNaN} from 'src/shared/utils/convertUserInput'
+
+// Components
+import DatePicker from 'src/shared/components/dateRangePicker/DatePicker'
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import {
+  Appearance,
+  Button,
+  ButtonRef,
+  Columns,
+  ComponentColor,
+  ComponentSize,
+  ComponentStatus,
+  Form,
+  Grid,
+  Input,
+  IconFont,
+  Popover,
+  PopoverInteraction,
+  PopoverPosition,
+} from '@influxdata/clockface'
+
+interface TimeTickInputProps {
+  axisName: string
+  tickOptions: string[]
+  initialTickOptionValue: number | string
+  label: string
+  placeholder?: string
+  setOptions: (optionName: string, arg: string[] | number) => void
+}
+
+export const TimeTickInput: FC<TimeTickInputProps> = props => {
+  const {
+    axisName,
+    tickOptions,
+    initialTickOptionValue,
+    label,
+    placeholder = 'RFC3339',
+    setOptions,
+  } = props
+
+  const [tickOptionInput, setTickOptionInput] = useOneWayState(
+    initialTickOptionValue === initialTickOptionValue
+      ? initialTickOptionValue
+      : ''
+  )
+
+  const [tickOptionInputStatus, setTickOptionInputStatus] = useState<
+    ComponentStatus
+  >(ComponentStatus.Default)
+
+  const [isDatePickerOpen, setIsDatePickerOpen] = useState<boolean>(false)
+
+  const [
+    isOnClickOutsideHandlerActive,
+    setIsOnClickOutsideHandlerActive,
+  ] = useState<boolean>(true)
+
+  const triggerRef: RefObject<ButtonRef> = createRef()
+
+  const updateTickOption = (value?: string | number) => {
+    const convertedValue = convertUserInputValueToNumOrNaN(
+      value === undefined ? tickOptionInput : value
+    )
+    const tickOptionNameWithoutAxis = label.split(' ').join('')
+    const tickOptionNameWithAxis = `${axisName.toLowerCase()}${tickOptionNameWithoutAxis}`
+    const filteredTickOptions = Array.isArray(tickOptions)
+      ? tickOptions.filter(option => option !== tickOptionNameWithAxis)
+      : []
+    setOptions(tickOptionNameWithoutAxis, convertedValue)
+    if (convertedValue === convertedValue) {
+      setOptions('GenerateAxisTicks', [
+        ...filteredTickOptions,
+        tickOptionNameWithAxis,
+      ])
+    } else {
+      setOptions('GenerateAxisTicks', filteredTickOptions)
+      if (tickOptionInput !== '') {
+        setTickOptionInputStatus(ComponentStatus.Error)
+      }
```

Comment:
Yes but with a small change. I will need to call `setOptions` with the computedTickOptions no matter what, so that call goes last.

And i think checking convertedValue might be confusing. I changed it to a check of `Number.isNaN` because that is basically what i want to know here.

Also, I changed it to an if - else if structure. i'd rather not check `NaN`ness twice. Just once and the `else if` can check the other condition.